src/main/java/reader/DataReader.java

```diff
@@ -90,4 +90,22 @@ public void writeFile(String fileContent, String path) {
       log.error("An error occurred.", e);
     }
   }
+
+  public void writeCsv(List<List<String>> fileContent, List<String> id, String path) {
+    try (FileWriter myWriter = new FileWriter(path)) {
+      int i = 0;
+      for (List<String> s : fileContent) {
+
+        for (String g : s) {
+          myWriter.write(g + "," + " " + "," + id.get(i));
+          myWriter.write("\n");
+        }
+        myWriter.write("\n\n");
+        i++;
```

Comment:
I think you can move the "logic " to another method in different class, so at the end you will first generate String that containes results foramted as CSV, and then you will pass this CSV String to  writeFile() method which you already have above.
writeFile method can be made ststic and be also moved to some kind of FileUtils class? what do you think?